main/res/values/strings.xml

```diff
@@ -252,6 +253,8 @@
     <string name="info_stored_image">New image saved to:</string>
     <string name="info_storing_static_maps">Trying to store static maps</string>
     <string name="info_cache_saved">The cache has been stored locally</string>
+    <string name="info_calculator_state_saved">Current state of the calculator has been saved to waypoint.</string>
```

Comment:
Hmm, yeah.  I see what I said above was a bit ambiguous.

To be clear, what we're talking about is the "Done" button in the top left corner:
![done](https://user-images.githubusercontent.com/6573522/28520265-f152ba0e-70ad-11e7-9f11-882b81277289.png)

Normally, when the user clicks the "Done" button, the coordinates are parsed and, if the result is valid, the result is used as the input for the way point and the calculator is closed.  If however the result is not valid a toast is displayed saying "c:geo can't parse latitude or longitude." and the calculator is left open so the user can correct the malformed coordinates and try again.  If the user doesn't want to repair the coordinates they can always click the "Cancel" cross to discard the malformed calculation and return to the waypoint with the values it had before —and this is exactly the same behavior as the CoordinateInputDialog itself uses.

The thing to workout is when should the calculators state itself be saved.  As it is now (which I think is the best way to have it) the calculators state is saved whenever the "Done" button is pressed.  The issue is that if the "Done" button is clicked while the coordinates are invalid the user is informed about the invalid coordinates but there is no indication about the calculators state being saved.

Another option would be to only save the calculators state if the coordinate's format is valid, however it is anticipated that people will often want to store partially computed waypoints and requiring that the coordinates be fully formatted before saving would be unnecessarily burdensome.

The final option would be to save the state every time either the "Done" tick OR the "Cancel" cross is clicked.  This would obviously mean though that a user would not be able cancel their erroneous scratchings and revert to their previously saved calculators state should they be uncertain about things, which doesn't seem like a good option.

There would also be options of allowing the back arrow to save the state and the "Cancel" button not to (or vice versa) but neither of these seem particularly natural and would not be intuitive to the user, and adding a dedicated "Save State" button or menu item seems like even more of a kludge.

After considering all these options (and even implementing most of them) I eventually settled on the "Always save when 'Done' is pressed" methodology and have elected to inform the user when this happens.  As it is the message is shown even when the coordinate format is valid but I actually think it's fine like that as the message is immediately followed by the "Be sure to save waypoint if you want to keep the calculator state." message once the 'Waypoint' screen is reached, which actually does look good.

If you can think of an approach that you believe would be better I'm open to suggestions but I would recommend at least having a go with the current implementation as it does seem pretty good to me.